### ISSUE TITLE
Typst will support  maybe ?

### ISSUE THREAD

**Issue by sgf** [MyNihongo/MudBlazor.Markdown#282]
https://github.com/typst/typst

**Comment by my-nihongo** [comment]
Sorry, but what exactly do you want to be supported?

**Comment by sgf** [comment]
Obviously, the Typst format is much better than the markdown format. But so far, any Typst repository in the .net open source community is still blank.

**Comment by sgf** [comment]
Obviously, the Typst format is much better than the markdown format. But so far, any Typst repository in the .net open source community is still blank.

**Comment by MihailsKuzmins** [comment]
I see, it might sound like a possible project, but definitely, it will be out of the scope of this repository (at least for the time being). I might say that it can be on the list of potential tasks to do, but probably now there are no parsers, so writing a custom one from scratch will also take some time, so unfortunately I cannot undertake this task due to other priorities. But when there is such a library, I would say it might sound fun.

So, I think I will keep this issue `open` and let's see how it goes.

**Comment by Foxtrek64** [comment]
Never heard of typst before so I looked into it. It appears to be a markup language but with more consideration for things like page width, headers and footers, etc.

I agree with Mihails. I think the best course of action is to wait for a general .NET library for typst, similar to how C# has a general Markdown library with Markdig. Once this library exists, we can look at writing a MudBlazor.Typst package or extending MudBlazor.Markdown to leverage this generic library.

If this is something you're interested in, implementing typst would be a good intermediate-level C# project. You could be the one to spearhead this project. I'd recommend looking at [Markdig](https://github.com/xoofx/markdig) and [QuestPDF](https://github.com/QuestPDF/QuestPDF) for inspiration.
